lib/step.rb

```diff
@@ -137,32 +135,55 @@ def run(dry_run: false, fail_fast: false)
 
         return if passed?
 
-        puts @output
-        puts
-
-        @named_args.each do |name|
-          next if name.blank?
+        if ENV["GITHUB_ACTIONS"].blank?
+          puts @output
+        else
+          os_string = if OS.linux?
+            "Linux"
+          elsif Hardware::CPU.arm?
+            "macOS #{MacOS.version.pretty_name} (#{MacOS.version}) on Apple Silicon"
+          else
+            "macOS #{MacOS.version.pretty_name}"
+          end
 
-          path, line = begin
-            formula = Formulary.factory(name)
-            method_sym = command.second.to_sym
-            method_location = formula.method(method_sym).source_location if formula.respond_to?(method_sym)
+          @named_args.each do |name|
+            if name.blank?
+              puts @output
+              next
+            end
 
-            if method_location.present? && (method_location.first == formula.path.to_s)
-              method_location
-            else
-              [formula.path, nil]
+            path, line = begin
+              formula = Formulary.factory(name)
+              method_sym = command.second.to_sym
+              method_location = formula.method(method_sym).source_location if formula.respond_to?(method_sym)
+
+              if method_location.present? && (method_location.first == formula.path.to_s)
+                method_location
+              else
+                [formula.path, nil]
+              end
+            rescue FormulaUnavailableError
+              [@repository.glob("**/#{name}*").first, nil]
             end
-          rescue FormulaUnavailableError
-            [@repository.glob("**/#{name}*").first, nil]
-          end
-          next if path.blank?
+            if path.blank?
+              puts @output
+              next
+            end
+
+            annotation_type = failed? ? :error : :warning
+
+            # GitHub Actions has a 64KB maximum for annotiations. That's a bit
+            # too long so instead let's go for a maximum of 256 lines or 24KB.
+            max_length_start = [@output.length - (24 * 1024), 0].max
+            annotation_output = @output.lines.last(100).join("\n")[max_length_start..]
```

Comment:
Won't `max_length_start` be the starting point relative to when `output` contains all the lines, and not when you start at line `-100`? What you actually want is `@output.lines.last(100).length - (24 * 1024)` I think.

I suggest you take the last x lines of `output` first and then compute `max_length_start`. `last(100)` also doesn't seem to match the comment which says `256` if I've not misunderstood what's happening here.

Reply:
Updated!